rstudio/rstudio
# use separate build library path when installing package

### Intent

Addresses https://github.com/rstudio/rstudio/issues/13399.

### Approach

On Windows, it's common for package installation to fail with errors of the form:

> ERROR: cannot remove earlier installation, is it in use?

typically because (as the error message states) the package is currently loaded; possibly in a totally separate R session. This PR seeks to alleviate the issue by:

- Installing the package into its own unique build library path,
- Performing the regular suspend + restart,
- On restart, we attempt to move the installed package to the intended install location,
- If that fails, we just add that directory to the library paths, and notify the user.

This way, even if the package is already in use by a separate R process, the user will still be able to install and test the package.

I'd like to add an option to enable / disable this behavior just in case.

### Automated Tests

N/A

### QA Notes

The simplest way to test on Windows would be to do something like the following:

1. In one RStudio session, run `library(rlang)` to load `rlang`;
2. In another RStudio session, create a new project from Git, and use https://github.com/r-lib/rlang as the repository location;
3. After the project is opened, try to Install the package.

You should see something like:

```
- RStudio was unable to move "rlang" into your package library.
- C:/Users/kevin/AppData/Local/R/win-library/4.4/build-fdbc1ec921b9 has been added to the library paths for this session.
> library(rlang)
```

### Documentation

N/A

### Checklist
